lib/Transforms/IPO/PassManagerBuilder.cpp

```diff
@@ -419,6 +419,7 @@ void PassManagerBuilder::populateModulePassManager(
     if (EnableMLSM)
       MPM.add(createMergedLoadStoreMotionPass()); // Merge ld/st in diamonds
     MPM.add(createGVNPass(DisableGVNLoadPRE));  // Remove redundancies
+    MPM.add(createDxilSimpleGVNHoistPass()); // HLSL Change - GVN hoist for code size.
```

Comment:
Let's add the -res_may_alias flag to dxc (already in fxc) and make sure the pass is disabled if this flag is set.